contracts/platform/registry/Registry.sol

```diff
@@ -70,16 +70,22 @@ contract Registry is Destructible, RegistryInterface, Upgradable {
         require(endpoint != bytes32(0), "Error: Can't initiate null endpoint");
 
         setCurve(msg.sender, endpoint, curve);
+
         db.pushBytesArray(keccak256(abi.encodePacked('oracles', msg.sender, 'endpoints')), endpoint);
-        db.setBytes32(keccak256(abi.encodePacked('oracles', msg.sender, endpoint, 'broker')), bytes32(broker));
+
+       
+        db.setBytes32(keccak256(abi.encodePacked('oracles', msg.sender, endpoint, 'broker')), bytes32(uint256(broker) << 96));
```

Comment:
compiler did not allow the casting of address to bytes32...

work around i found online was to cast the address to uint256 then to bytes32